tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-loader.js

```diff
@@ -167,11 +162,34 @@ Polymer({
    * @private
    */
   _updateScene: function(currentStep) {
-    this._meshViewer.updateScene(currentStep, this);
-    if (!this._cameraPositionInitialized) {
-      this._meshViewer.resetView();
-      this._cameraPositionInitialized = true;
+    let dataPromise = Promise.resolve();
```

Comment:
AFAICT, `currentStep` can change quite rapidly: setting slider aside, `reload` sets `steps` and `stepIndex` in two steps causing `currentStep` to change twice in rapid succession. Can you make sure we are not fetching the same mesh few times?

Reply:
Mmm, interesting behavior. But it is going to happen only once, when page is loaded, right? Should we complicate the logic for such small optimization?

Reply:
> But it is going to happen only once, when page is loaded, right?

It looks like this method gets triggered every time `currentStep` changes and fetch happens if no mesh is available for that step. `currentStep` can change when user uses the slider to change the `stepIndex`.

Reply:
When you say `currentStep` can change when user uses the slider to change the `stepIndex`. Do you mean that `currentStep` object can be replaced?